packages/dbml-core/src/parse/dbml/src/services/suggestions.ts

```diff
@@ -0,0 +1,731 @@
+          break;
+        }
+        settingNameFragments.push(token.value);
+        iter = iter.back();
+      }
+
+      const res = this.suggestAttributeValue(model, offset, settingNameFragments.join(' '));
+
+      return !shouldAppendSpace(colon, offset) ? res : prependSpace(res);
+    }
+
+    if (!ctx) {
+      return noSuggestions();
+    }
+
+    const lineIter = TokenLogicalLineIterator.fromOffset(this.compiler, offset);
+    // Note: An incomplete simple element declaration may corrupt the ElementDeclarationNode
+    // and `ctx` may fail to hold the precise scope, rather, it would holds the parent's scope
+    // e.g
+    // Ref R: // At this point, the scope would be `TopLevel`
+    // Therefore, this check also tries to inspect previous tokens (before :)
+    if (ctx.scope?.kind === ScopeKind.REF || isAtStartOfSimpleBody(lineIter, ElementKind.REF)) {
+      const res = this.suggestNamesInScope(model, offset, ctx.element?.node, [
+        SymbolKind.Schema,
+        SymbolKind.Table,
+      ]);
+
+      return !shouldAppendSpace(colon, offset) ? res : prependSpace(res);
+    }
+
+    return noSuggestions();
+  }
+
+  private suggestAttributeValue(
+    model: TextModel,
+    offset: number,
+    settingName: string,
+  ): CompletionList {
+    switch (settingName?.toLowerCase()) {
+      case 'update':
+      case 'delete':
+        return {
+          suggestions: ['cascade', 'set default', 'set null', 'restrict'].map((name) => ({
+            label: name,
+            insertText: name,
+            kind: CompletionItemKind.Value,
+            insertTextRules: CompletionItemInsertTextRule.KeepWhitespace,
+            range: undefined as any,
+          })),
+        };
+      case 'type':
+        return {
+          suggestions: ['btree', 'hash'].map((name) => ({
+            label: name,
+            insertText: `${name}`,
+            kind: CompletionItemKind.Value,
+            insertTextRules: CompletionItemInsertTextRule.KeepWhitespace,
+            range: undefined as any,
+          })),
+        };
+      default:
+        break;
+    }
+
+    return noSuggestions();
+  }
+
+  private suggestMembers(
+    model: TextModel,
+    offset: number,
+    iterFromDot: TokenSourceIterator,
+  ): CompletionList {
+    const nameStack: string[] = [];
+    let curIter = iterFromDot;
+    let curToken = iterFromDot.value().unwrap_or(undefined);
+
+    while (isDot(curToken)) {
+      curIter = curIter.back();
+      curToken = curIter.value().unwrap_or(undefined);
+      if (
+        curToken?.kind === SyntaxTokenKind.IDENTIFIER ||
+        curToken?.kind === SyntaxTokenKind.QUOTED_STRING
+      ) {
+        nameStack.unshift(curToken.value);
+      } else {
+        return noSuggestions();
+      }
+      curIter = curIter.back();
+      curToken = curIter.value().unwrap_or(undefined);
+    }
+
+    return {
+      suggestions: this.compiler.membersOfName(nameStack).map(({ kind, name }) => ({
```

Comment:
Why not doing like this, take this schema property access expression as an example: `v1.users.[suggestions]`:

- Finding the name stack
- Get the node of the name `v1` (call the `.containers(offset)`, get the most inner node for example)
- Get symbol of this node (`node.symbol`)
- Then we resolve the member's chain recursively (v1 -> users -> [suggestions])

instead of guessing and using the current ast's symbol table in `ofName`? In the future, if the symbol of this node is declared in different files, using the current ast's symbol table is no longer correct.